client.go

```diff
@@ -110,10 +110,10 @@ func NewRequest(method, url string, rawBody interface{}) (*Request, error) {
 	var contentLength int64
 
 	if rawBody != nil {
-		switch rawBody.(type) {
+		switch rawBody := rawBody.(type) {
```

Comment:
To make it clearer what's happening here (so people's minds don't miss the scope change even if the compiler doesn't), can you make the variable you're assigning to named something else?